lib/rules/utils/ast-utils.js

```diff
@@ -464,14 +464,24 @@ function isArrayFromMethod(node) {
  * @returns {boolean} Whether or not the node is a method which has `thisArg`.
  */
 function isMethodWhichHasThisArg(node) {
-    for (
-        let currentNode = node;
-        currentNode.type === "MemberExpression" && !currentNode.computed;
-        currentNode = currentNode.property
-    ) {
-        if (currentNode.property.type === "Identifier") {
-            return arrayMethodPattern.test(currentNode.property.name);
+    let currentNode = node;
+
+    while (currentNode) {
+        if (currentNode.type === "ChainExpression") {
+            currentNode = currentNode.expression;
+            continue;
         }
+
+        if (currentNode.type === "MemberExpression" && !currentNode.computed) {
+            if (currentNode.property.type === "Identifier") {
+                return arrayMethodPattern.test(currentNode.property.name);
+            }
+
+            currentNode = currentNode.property;
```

Comment:
I still think this loop is unnecessary, we're checking only one `property`. Also, `currentNode.property` cannot be `MemberExpression` at this point, so there will be no useful iterations afterward.

Reply:
Really. I'm not sure what I thought.